### PR TITLE
create sum_of_digits.py

### PR DESCRIPTION
**Pull request by vicky1999**
create sum_of_digits.py to find the sum of digits of a number
digit_sum(12345) ---> 15
digit_sum(12345) ---> 10

### CODE REVIEW

File: maths/sum_of_digits.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+def digit_sum(n: int) -> int:
```
Comment (cclauss) [inline maths/sum_of_digits.py:1]:
```suggestion
def sum_of_digits(n: int) -> int:
```
Seems like a more self documenting name.

File: maths/sum_of_digits.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Find the sum of digits of a number.
+
+    >>> digit_sum(12345)
+    15
+    >>> digit_sum(123)
+    6
+    """
+    res = 0
+    while n > 0:
+        res += n % 10
```
Comment (cclauss) [inline maths/sum_of_digits.py:12]: Could we optimize using Python builtin https://docs.python.org/3/library/functions.html#divmod ?

File: maths/sum_of_digits.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        n = n // 10
+    return res
+
+
+if __name__ == "__main__":
+    print(digit_sum(12345))   # ===> 15
```
Comment (cclauss) [inline maths/sum_of_digits.py:18]:
```suggestion
    print(digit_sum(12345))  # ===> 15
```

File: maths/sum_of_digits.py (hunk 2 above)
Comment (vicky1999) [reply maths/sum_of_digits.py:12]: Yeah, divmod() method would be faster than mod operator.  But for small values, there is not much difference. Everytime, we mod a number with 10 until it becomes 0 will not have much effect on the running time I guess.

Comment (cclauss) [reply maths/sum_of_digits.py:12]:
```python
def digits(n):
        while n:
            n, digit = divmod(n, 10)
            yield digit
         
print(sum(digits(12345)))
```

Comment (vicky1999) [reply maths/sum_of_digits.py:12]: Yeah, that's the digit sum function using divmod()

**Pull request merged**